code/game/machinery/OpTable.dm

```diff
@@ -43,60 +40,63 @@
 		return TRUE
 
 /obj/machinery/optable/CanPass(atom/movable/mover, turf/target, height=0)
-	if(height==0) return 1
-
+	if(height==0)
+		return TRUE
 	if(istype(mover) && mover.checkpass(PASSTABLE))
-		return 1
+		return TRUE
 	else
-		return 0
+		return FALSE
 
 
-/obj/machinery/optable/MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
-	if(usr.stat || (!ishuman(user) && !isrobot(user)) || user.restrained() || !check_table(user) || user.IsWeakened() || user.stunned)
+/obj/machinery/optable/MouseDrop_T(atom/movable/O as mob|obj, var/mob/P)
+	if(usr.stat || (!ishuman(P) && !isrobot(P)) || !check_table())
 		return
-
-	if(!ismob(O)) //humans only
+	if(!check_table()) //Is the Operating Table currently occupied?
+		return
+	if(P.restrained() || P.buckled || P.incapacitated()) //Folks who are restrained or otherwise incapacitated cannot get on the table
+		return
+	if(!ismob(O)) //Humans only
 		return
-
 	if(istype(O, /mob/living/simple_animal) || istype(O, /mob/living/silicon)) //animals and robots dont fit
 		return
-
 	var/mob/living/L = O
-	take_victim(L,usr)
+	take_patient(L, usr)
```

Comment:
```dm
var/mob/living/carbon/C = O
if(!C)  // humanoids only
  return
take_patient(C, usr)
```
The `ismob()` check does not do what the previous coder expected it to do - it would match all the weird movable things (like ghosts) (and would likely runtime later).
The *intent* here seems to only be taking in humanoids, so let's just check for that instead of matching everything and special-casing some of the things we don't want.

Reply:
Thanks! I was confused by this too!

Reply:
`ishuman(O)`